supabase/functions/cron_email/index.ts

```diff
@@ -0,0 +1,156 @@
+import { serve } from 'https://deno.land/std@0.199.0/http/server.ts'
+import { getEnv, methodJson, sendRes } from '../_utils/utils.ts'
+import type { BaseHeaders } from '../_utils/types.ts'
+import { trackEvent } from '../_utils/plunk.ts'
+import { supabaseAdmin } from '../_utils/supabase.ts'
+
+// This is required propoably due to this https://github.com/supabase/postgrest-js/issues/408
+interface AppWithUser {
+  app_id: string
+  name: string | null
+  user_id: {
+    email: string
+  }
+}
+
+const thresholds = {
+  // Number of updates in plain number
+  updates: [
+    100,
+    1000,
+    10000,
+  ],
+  // Percentage in decimal form (0.9 ==== 90%)
+  failRate: [
+    0.80,
+    0.90,
+    0.95,
+  ],
+  // Number of app opens in plain number
+  appOpen: [
+    500,
+    1500,
+    5000,
+  ],
+}
+
+const funComparisons = {
+  updates: [
+    'a cupcake to every student in a small school!',
+    'a pizza to every resident of a small town!',
+    'a burger to everyone in a big city!',
+  ],
+  failRate: [
+    'Even cats don\'t land on their feet that often!',
+    'That\'s a success rate higher than the average pass rate of a tough university exam!',
+    'That\'s a success rate that even the best basketball players would envy!',
+  ],
+  appOpen: [
+    'Your app was opened more times than a popular local bakery\'s door!',
+    'Your app was more popular than the latest episode of a hit TV show!',
+    'Your app was opened more times than a blockbuster movie on its opening weekend!',
+  ],
+}
+
+// Check what treshold does the stat qualify for and return the fun comparison
+function getFunComparison(comparison: keyof typeof funComparisons, stat: number): string {
+  const thresholdsForComparisons = thresholds[comparison]
+  const index = thresholdsForComparisons.findIndex((threshold, index) => {
+    if (threshold >= stat)
+      return true
+    else if ((index === 2 && stat >= threshold))
+      return true // Last index
+
+    return false
+  })
```

Comment:
I know that I already gave a suggestion here, but I guess this would give the same result

```suggestion
  const index = thresholdsForComparisons.findIndex((threshold) => threshold >= stat)
```

Reply:
No it will not, for example if you had 10 000 app openings this week then the last threshold would we `5000` and `10 000` is bigger then `5000` thus index would not be found

Reply:
oh, I didn't see the second one was inverted. sorry

we could simplify it and clarify what these conditions mean (I'm pretty new to the project, and the conditions don't tell me what this code does.)

what if we can do something like
```suggestion
  const index = thresholdsForComparisons.findIndex((threshold, index) => {
    const firstCondition = threshold >= stat;
    const secondCondition = index === 2 && stat >= threshold;

    return firstCondition || secondCondition;
  })
```

and change the variables name to explicitly (`thresholdIsSomething`, `thresholdMatchXCondition`); what do you think?

Reply:
Rename `firstCondition` to `thresholdGreaterThenStat` and to `secondCondition` to `statGreaterOrEqualThresholdAndLastIndex`, suggest changes and I will commit this

Reply:
```suggestion
  const index = thresholdsForComparisons.findIndex((threshold, index) => {
    const thresholdGreaterThenStat = threshold >= stat;
    const statGreaterOrEqualThresholdAndLastIndex = index === 2 && stat >= threshold;

    return thresholdGreaterThenStat || statGreaterOrEqualThresholdAndLastIndex;
  })
```

thanks for accepting my suggestion 😄